### ISSUE TITLE
Range.abuts() is true when it shouldn't

### ISSUE THREAD

**Issue by matheus220** [LaNoC-UFC/rsbr#97]
The order of the parameters of Range.abuts() method influences the results in some cases.

**Comment by jmferreiratech** [comment]
@matheus220 could you please give us an example in which that happens?

**Comment by matheus220** [comment]
@jmarcelof, we can use the [TwoDimensionRangeTest.AbutsRight()](https://github.com/LaNoC-UFC/rsbr/blob/develop/tests/TwoDimensionRangeTest.java#L79) to check for two cases:

**CASE 1:**
```
public void AbutsRight() throws Exception {
        Range range1 = Range.TwoDimensionalRange(1, 1, 0, 0);
        Range range2 = Range.TwoDimensionalRange(1, 1, 3, 3);
        Assert.assertTrue(range1.abuts(range2));
}
```

**CASE 2:**
```
public void AbutsRight() throws Exception {
        Range range1 = Range.TwoDimensionalRange(1, 1, 0, 0);
        Range range2 = Range.TwoDimensionalRange(1, 1, 3, 3);
        Assert.assertTrue(range2.abuts(range1));
}
```

When doing the tests we get different results, one correct and the other not.

**Comment by matheus220** [comment]
I suggest this change, what do you think?

**Comment by jmferreiratech** [comment]
@matheus220, now that I saw your examples a noted that you found a different bug. Neither of the cases you described should pass, A point at (1, 0) does not abut with one at (1, 3). Am I right @rafaelgmota?
Anyway, the patch has to contain a unit test that do not pass with that bug alongside with the fix.

**Comment by matheus220** [comment]
@jmarcelof, correct, neither case should pass. That's the problem, because one of the cases is passing. Am I right?

**Comment by jmferreiratech** [comment]
I think that's fixed by #99. Please review it and let me know if that is right.
